File: app/src/main/java/com/jann_luellmann/thekenapp/data/dao/EventCustomerCrossDAO.java
package com.jann_luellmann.thekenapp.data.dao;

import com.jann_luellmann.thekenapp.data.model.EventCustomerCrossRef;
import com.jann_luellmann.thekenapp.data.model.relationship.EventWithCustomers;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Transaction;

@Dao
public interface EventCustomerCrossDAO {

    @Transaction
    @Query("SELECT * FROM eventCustomerCrossRef")
    LiveData<List<EventCustomerCrossRef>> getAll();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertAll(EventCustomerCrossRef... eventCustomerCrossRef);

    @Delete
    void delete(EventCustomerCrossRef eventWithCustomers);
}

File: app/src/main/java/com/jann_luellmann/thekenapp/data/dao/EventCustomerCrossDAO.kt
package com.jann_luellmann.thekenapp.data.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.jann_luellmann.thekenapp.data.model.EventCustomerCrossRef
import com.jann_luellmann.thekenapp.data.model.EventDrinkCrossRef

@Dao
interface EventCustomerCrossDAO {
    @get:Query("SELECT * FROM eventCustomerCrossRef")
    @get:Transaction
    val all: LiveData<List<EventCustomerCrossRef>>

    @Query("SELECT * FROM eventCustomerCrossRef WHERE eventId = :eventId")
    fun findAllById(eventId: Long) : LiveData<List<EventCustomerCrossRef>>

    @Query("SELECT * FROM eventCustomerCrossRef WHERE eventId = :eventId")
    fun findAllByIdBlocking(eventId: Long) : List<EventCustomerCrossRef>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertAll(vararg eventCustomerCrossRef: EventCustomerCrossRef)

    @Delete
    fun delete(eventWithCustomers: EventCustomerCrossRef)
}